piecedirectory/piecedirectory.go

```diff
@@ -292,12 +294,35 @@ func (ps *PieceDirectory) addIndexForPiece(ctx context.Context, pieceCid cid.Cid
 	if !errors.Is(err, io.EOF) {
 		return fmt.Errorf("generating index for piece %s: %w", pieceCid, err)
 	}
+	concurrency := AddIndexConcurrency
+	// in an unlikely case if there are less than 8 indexes
+	if concurrency > len(recs) {
+		concurrency = len(recs)
+	}
+
+	chunkSize := len(recs) / concurrency
+	eg := errgroup.Group{}
 
-	// Add mh => piece index to store: "which piece contains the multihash?"
-	// Add mh => offset index to store: "what is the offset of the multihash within the piece?"
-	log.Debugw("add index: store index in local index directory", "pieceCid", pieceCid)
-	if err := ps.store.AddIndex(ctx, pieceCid, recs, true); err != nil {
-		return fmt.Errorf("adding CAR index for piece %s: %w", pieceCid, err)
+	for i := 0; i < concurrency; i++ {
+		idx := i
```

Comment:
Nit, but you can just continue to use `i` here, as in `i := i`, no need to introduce another name.